api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -740,7 +751,249 @@ public void shouldTestPurgeProgramAttributeType() throws Exception {
                 int totalAttributeTypes = pws.getAllProgramAttributeTypes().size();
 		pws.purgeProgramAttributeType(programAttributeType);
 		assertEquals((totalAttributeTypes - 1), pws.getAllProgramAttributeTypes().size());
-}
+	}
+	
+	
+	@Test
+	public void getPrograms_shouldTestGetPrograms() {
+		List<Program> programs = pws.getPrograms("MAL");
+		List<Program> programs1 = pws.getPrograms("PR");
+		assertEquals(programs.size(), 1);
+		assertEquals(programs1.size(), 3);
+	}
+	
+	@Test
+	public void retireProgram_shouldTestRetireProgram() {
+		Program programTest = pws.getAllPrograms().get(0);
+		Program program = pws.retireProgram(programTest, "expired");
+		for (ProgramWorkflow workflow : program.getWorkflows()) {
+			assertEquals(workflow.getRetired(), true);
```

Comment:
shouldn't you also need to check whether `workflow.getStates()` is not empty?

Reply:
yes, I added this